### ISSUE TITLE
db-less kong 1.2.1 config reload doesn't take effect

### ISSUE THREAD

**Issue by luxurine** [Kong/kong#4808]
### Summary
kong 1.2.1 db-less mode
use `kong reload` to update config(change target), the config changed, but not take effect, request still got response from old target, and the staus of new target is 'DNS_ERROR'

it seems the balancer doesn't rebuilt

### Steps To Reproduce

1、start kong with config 1-kong.yaml
ln -sf 1-kong.yaml
kong start
curl 127.0.0.1/ping

access log:
{"@timestamp":"2019-07-18T13:57:15+08:00","host":"127.0.0.1","clientip":"127.0.0.1","http_x_forwarded_for":"-","status":"200","request_method":"GET","request_url":"/ping","request_time":"0.001","size":"5","upstream_time":"0.001","upstream_host":"10.16.67.201:9094","server_name":"127.0.0.1","uri":"/ping","http_referer":"-","http_user_agent":"curl/7.29.0"}

http :8001/upstreams/592f0298-52d7-58ff-8bb4-74bc9e32e0c0/health
{
    "data": [
        {
            "created_at": 1563430327.428,
            "health": "HEALTHY",
            "id": "7af2b348-94cb-5828-a0c9-0a4febb14b4c",
            "target": "10.16.67.201:9094",
            "upstream": {
                "id": "592f0298-52d7-58ff-8bb4-74bc9e32e0c0"
            },
            "weight": 100
        }
    ],
    "next": null,
    "node_id": "b19a9ed5-1d7b-4539-9384-4faac0edd8c1"
}

2、switch config 2-kong.yaml
ln -sf 2-kong.yaml kong.yaml
kong reload

access log (still previous one):
{"@timestamp":"2019-07-18T13:57:15+08:00","host":"127.0.0.1","clientip":"127.0.0.1","http_x_forwarded_for":"-","status":"200","request_method":"GET","request_url":"/ping","request_time":"0.001","size":"5","upstream_time":"0.001","upstream_host":"10.16.67.201:9094","server_name":"127.0.0.1","uri":"/ping","http_referer":"-","http_user_agent":"curl/7.29.0"}

http :8001/upstreams/592f0298-52d7-58ff-8bb4-74bc9e32e0c0/health
{
    "data": [
        {
            "created_at": 1563430451.187,
            "health": "DNS_ERROR",
            "id": "6538d995-1b47-5068-a7ca-fd3df6718d1c",
            "target": "10.16.67.189:9001",
            "upstream": {
                "id": "592f0298-52d7-58ff-8bb4-74bc9e32e0c0"
            },
            "weight": 100
        }
    ],
    "next": null,
    "node_id": "b19a9ed5-1d7b-4539-9384-4faac0edd8c1"
}

```yaml
# cat 1-kong.yaml
_format_version: '1.1'
services:
- connect_timeout: 60000
  host: testsant
  name: testsant
  plugins:
  - config: {}
    enabled: true
    name: prometheus
    protocols:
    - http
    - https
    run_on: first
    tags: null
  port: 8080
  protocol: http
  read_timeout: 60000
  retries: 0
  routes:
  - hosts:
    - 127.0.0.1
    methods: []
    name: testsant
    plugins: []
    preserve_host: true
    protocols:
    - http
    - https
    regex_priority: 10
    strip_path: false
    tags: null
  write_timeout: 60000
upstreams:
- hash_fallback: none
  hash_on: none
  hash_on_cookie_path: /
  healthchecks:
    active:
      concurrency: 100
      healthy:
        http_statuses:
        - 200
        - 302
        interval: 1
        successes: 3
      http_path: /ping
      https_verify_certificate: true
      timeout: 1
      type: http
      unhealthy:
        http_failures: 3
        http_statuses:
        - 429
        - 404
        - 500
        - 501
        - 502
        - 503
        - 504
        - 505
        interval: 1
        tcp_failures: 3
        timeouts: 3
    passive:
      healthy:
        http_statuses:
        - 200
        - 201
        - 202
        - 203
        - 204
        - 205
        - 206
        - 207
        - 208
        - 226
        - 300
        - 301
        - 302
        - 303
        - 304
        - 305
        - 306
        - 307
        - 308
        successes: 0
      type: http
      unhealthy:
        http_failures: 0
        http_statuses:
        - 429
        - 500
        - 503
        tcp_failures: 0
        timeouts: 0
  name: testsant
  slots: 100
  tags: null
  targets:
  - target: 10.16.67.201:9094
    weight: 100
```

```yaml
# cat 2-kong.yaml
_format_version: '1.1'
services:
- connect_timeout: 60000
  host: testsant
  name: testsant
  plugins:
  - config: {}
    enabled: true
    name: prometheus
    protocols:
    - http
    - https
    run_on: first
    tags: null
  port: 8080
  protocol: http
  read_timeout: 60000
  retries: 0
  routes:
  - hosts:
    - 127.0.0.1
    methods: []
    name: testsant
    plugins: []
    preserve_host: true
    protocols:
    - http
    - https
    regex_priority: 10
    strip_path: false
    tags: null
  write_timeout: 60000
upstreams:
- hash_fallback: none
  hash_on: none
  hash_on_cookie_path: /
  healthchecks:
    active:
      concurrency: 100
      healthy:
        http_statuses:
        - 200
        - 302
        interval: 1
        successes: 3
      http_path: /ping
      https_verify_certificate: true
      timeout: 1
      type: http
      unhealthy:
        http_failures: 3
        http_statuses:
        - 429
        - 404
        - 500
        - 501
        - 502
        - 503
        - 504
        - 505
        interval: 1
        tcp_failures: 3
        timeouts: 3
    passive:
      healthy:
        http_statuses:
        - 200
        - 201
        - 202
        - 203
        - 204
        - 205
        - 206
        - 207
        - 208
        - 226
        - 300
        - 301
        - 302
        - 303
        - 304
        - 305
        - 306
        - 307
        - 308
        successes: 0
      type: http
      unhealthy:
        http_failures: 0
        http_statuses:
        - 429
        - 500
        - 503
        tcp_failures: 0
        timeouts: 0
  name: testsant
  slots: 100
  tags: null
  targets:
  - target: 10.16.67.189:9001
    weight: 100
```

### Additional Details & Logs

- Kong version : 1.2.1 
- Kong debug-level startup logs
# /usr/local/bin/kong start --vv
2019/07/18 14:02:10 [verbose] Kong: 1.2.1
2019/07/18 14:02:10 [debug] ngx_lua: 10013
2019/07/18 14:02:10 [debug] nginx: 1013006
2019/07/18 14:02:10 [debug] Lua: LuaJIT 2.1.0-beta3
2019/07/18 14:02:10 [verbose] reading config file at /etc/kong/kong.conf
2019/07/18 14:02:10 [debug] reading environment variables
2019/07/18 14:02:10 [debug] admin_access_log = "/data/logs/kong/admin_access.log"
2019/07/18 14:02:10 [debug] admin_error_log = "/data/logs/kong/error.log"
2019/07/18 14:02:10 [debug] admin_listen = {"0.0.0.0:8001"}
2019/07/18 14:02:10 [debug] anonymous_reports = false
2019/07/18 14:02:10 [debug] cassandra_consistency = "ONE"
2019/07/18 14:02:10 [debug] cassandra_contact_points = {"127.0.0.1"}
2019/07/18 14:02:10 [debug] cassandra_data_centers = {"dc1:2","dc2:3"}
2019/07/18 14:02:10 [debug] cassandra_keyspace = "kong"
2019/07/18 14:02:10 [debug] cassandra_lb_policy = "RequestRoundRobin"
2019/07/18 14:02:10 [debug] cassandra_port = 9042
2019/07/18 14:02:10 [debug] cassandra_repl_factor = 1
2019/07/18 14:02:10 [debug] cassandra_repl_strategy = "SimpleStrategy"
2019/07/18 14:02:10 [debug] cassandra_schema_consensus_timeout = 10000
2019/07/18 14:02:10 [debug] cassandra_ssl = false
2019/07/18 14:02:10 [debug] cassandra_ssl_verify = false
2019/07/18 14:02:10 [debug] cassandra_timeout = 5000
2019/07/18 14:02:10 [debug] cassandra_username = "kong"
2019/07/18 14:02:10 [debug] client_body_buffer_size = "64k"
2019/07/18 14:02:10 [debug] client_max_body_size = "0"
2019/07/18 14:02:10 [debug] client_ssl = false
2019/07/18 14:02:10 [debug] database = "off"
2019/07/18 14:02:10 [debug] db_cache_ttl = 0
2019/07/18 14:02:10 [debug] db_cache_warmup_entities = {"services","plugins"}
2019/07/18 14:02:10 [debug] db_resurrect_ttl = 30
2019/07/18 14:02:10 [debug] db_update_frequency = 5
2019/07/18 14:02:10 [debug] db_update_propagation = 0
2019/07/18 14:02:10 [debug] declarative_config = "/etc/kong/yaml/kong.yaml"
2019/07/18 14:02:10 [debug] dns_error_ttl = 1
2019/07/18 14:02:10 [debug] dns_hostsfile = "/etc/hosts"
2019/07/18 14:02:10 [debug] dns_no_sync = false
2019/07/18 14:02:10 [debug] dns_not_found_ttl = 30
2019/07/18 14:02:10 [debug] dns_order = {"LAST","SRV","A","CNAME"}
2019/07/18 14:02:10 [debug] dns_resolver = {}
2019/07/18 14:02:10 [debug] dns_stale_ttl = 4
2019/07/18 14:02:10 [debug] error_default_type = "text/plain"
2019/07/18 14:02:10 [debug] headers = {"server_tokens","latency_tokens"}
2019/07/18 14:02:10 [debug] log_level = "notice"
2019/07/18 14:02:10 [debug] lua_package_cpath = ""
2019/07/18 14:02:10 [debug] lua_package_path = "./?.lua;./?/init.lua;"
2019/07/18 14:02:10 [debug] lua_socket_pool_size = 30
2019/07/18 14:02:10 [debug] lua_ssl_verify_depth = 1
2019/07/18 14:02:10 [debug] mem_cache_size = "256m"
2019/07/18 14:02:10 [debug] nginx_admin_directives = {}
2019/07/18 14:02:10 [debug] nginx_daemon = "on"
2019/07/18 14:02:10 [debug] nginx_http_directives = {{value="on",name="tcp_nodelay"},{value="json '{\"@timestamp\":\"$time_iso8601\",\"host\":\"$server_addr\",\"clientip\":\"$remote_addr\",\"http_x_forwarded_for\":\"$http_x_forwarded_for\",\"status\":\"$status\",\"request_method\":\"$request_method\",\"request_url\":\"$request_uri\",\"request_time\":\"$request_time\",\"size\":\"$body_bytes_sent\",\"upstream_time\":\"$upstream_response_time\",\"upstream_host\":\"$upstream_addr\",\"server_name\":\"$host\",\"uri\":\"$uri\",\"http_referer\":\"$http_referer\",\"http_user_agent\":\"$http_user_agent\"}'",name="log_format"},{value="\"1000\"",name="keepalive_requests"},{value="60s",name="keepalive_timeout"},{value="prometheus_metrics 100m",name="lua_shared_dict"}}
2019/07/18 14:02:10 [debug] nginx_http_keepalive_requests = "1000"
2019/07/18 14:02:10 [debug] nginx_http_keepalive_timeout = "60s"
2019/07/18 14:02:10 [debug] nginx_http_log_format = "json '{\"@timestamp\":\"$time_iso8601\",\"host\":\"$server_addr\",\"clientip\":\"$remote_addr\",\"http_x_forwarded_for\":\"$http_x_forwarded_for\",\"status\":\"$status\",\"request_method\":\"$request_method\",\"request_url\":\"$request_uri\",\"request_time\":\"$request_time\",\"size\":\"$body_bytes_sent\",\"upstream_time\":\"$upstream_response_time\",\"upstream_host\":\"$upstream_addr\",\"server_name\":\"$host\",\"uri\":\"$uri\",\"http_referer\":\"$http_referer\",\"http_user_agent\":\"$http_user_agent\"}'"
2019/07/18 14:02:10 [debug] nginx_http_lua_shared_dict = "prometheus_metrics 100m"
2019/07/18 14:02:10 [debug] nginx_http_tcp_nodelay = "on"
2019/07/18 14:02:10 [debug] nginx_optimizations = true
2019/07/18 14:02:10 [debug] nginx_proxy_directives = {}
2019/07/18 14:02:10 [debug] nginx_sproxy_directives = {}
2019/07/18 14:02:10 [debug] nginx_stream_directives = {}
2019/07/18 14:02:10 [debug] nginx_user = "nobody nobody"
2019/07/18 14:02:10 [debug] nginx_worker_processes = "auto"
2019/07/18 14:02:10 [debug] origins = {}
2019/07/18 14:02:10 [debug] pg_database = "kong"
2019/07/18 14:02:10 [debug] pg_host = "127.0.0.1"
2019/07/18 14:02:10 [debug] pg_max_concurrent_queries = 0
2019/07/18 14:02:10 [debug] pg_port = 5432
2019/07/18 14:02:10 [debug] pg_semaphore_timeout = 60000
2019/07/18 14:02:10 [debug] pg_ssl = false
2019/07/18 14:02:10 [debug] pg_ssl_verify = false
2019/07/18 14:02:10 [debug] pg_timeout = 5000
2019/07/18 14:02:10 [debug] pg_user = "kong"
2019/07/18 14:02:10 [debug] plugins = {"ip-restriction","prometheus","rate-limiting","file-log"}
2019/07/18 14:02:10 [debug] prefix = "/usr/local/kong/"
2019/07/18 14:02:10 [debug] proxy_access_log = "/data/logs/kong/access.log json"
2019/07/18 14:02:10 [debug] proxy_error_log = "/data/logs/kong/error.log"
2019/07/18 14:02:10 [debug] proxy_listen = {"0.0.0.0:80","0.0.0.0:443 ssl"}
2019/07/18 14:02:10 [debug] real_ip_header = "X-Real-IP"
2019/07/18 14:02:10 [debug] real_ip_recursive = "off"
2019/07/18 14:02:10 [debug] router_consistency = "strict"
2019/07/18 14:02:10 [debug] ssl_cipher_suite = "custom"
2019/07/18 14:02:10 [debug] ssl_ciphers = "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE:ECDH:AES:HIGH:!NULL:!aNULL:!MD5:!ADH:!RC4:!DH:!DHE"
2019/07/18 14:02:10 [debug] stream_listen = {"off"}
2019/07/18 14:02:10 [debug] trusted_ips = {}
2019/07/18 14:02:10 [debug] upstream_keepalive = 2048
2019/07/18 14:02:10 [verbose] prefix in use: /usr/local/kong
2019/07/18 14:02:10 [debug] loading subsystems migrations...
2019/07/18 14:02:10 [verbose] retrieving cache schema state...
2019/07/18 14:02:10 [verbose] schema state retrieved
2019/07/18 14:02:10 [verbose] preparing nginx prefix directory at /usr/local/kong
2019/07/18 14:02:10 [verbose] SSL enabled, no custom certificate set: using default certificate
2019/07/18 14:02:10 [verbose] default SSL certificate found at /usr/local/kong/ssl/kong-default.crt
2019/07/18 14:02:10 [debug] searching for OpenResty 'nginx' executable
2019/07/18 14:02:10 [debug] /usr/local/openresty/nginx/sbin/nginx -v: 'nginx version: openresty/1.13.6.2'
2019/07/18 14:02:10 [debug] found OpenResty 'nginx' executable at /usr/local/openresty/nginx/sbin/nginx
2019/07/18 14:02:10 [debug] testing nginx configuration: KONG_NGINX_CONF_CHECK=true /usr/local/openresty/nginx/sbin/nginx -t -p /usr/local/kong -c nginx.conf
2019/07/18 14:02:10 [debug] searching for OpenResty 'nginx' executable
2019/07/18 14:02:10 [debug] /usr/local/openresty/nginx/sbin/nginx -v: 'nginx version: openresty/1.13.6.2'
2019/07/18 14:02:10 [debug] found OpenResty 'nginx' executable at /usr/local/openresty/nginx/sbin/nginx
2019/07/18 14:02:10 [debug] sending signal to pid at: /usr/local/kong/pids/nginx.pid
2019/07/18 14:02:10 [debug] kill -0 `cat /usr/local/kong/pids/nginx.pid` >/dev/null 2>&1
2019/07/18 14:02:10 [debug] starting nginx: /usr/local/openresty/nginx/sbin/nginx -p /usr/local/kong -c nginx.conf
2019/07/18 14:02:10 [debug] nginx started
2019/07/18 14:02:10 [info] Kong started

- Kong error logs:
 none
- Kong configuration
{
    "configuration": {
        "admin_acc_logs": "/usr/local/kong/logs/admin_access.log",
        "admin_access_log": "/data/logs/kong/admin_access.log",
        "admin_error_log": "/data/logs/kong/error.log",
        "admin_listen": [
            "0.0.0.0:8001"
        ],
        "admin_listeners": [
            {
                "http2": false,
                "ip": "0.0.0.0",
                "listener": "0.0.0.0:8001",
                "port": 8001,
                "proxy_protocol": false,
                "ssl": false,
                "transparent": false
            }
        ],
        "admin_ssl_cert_default": "/usr/local/kong/ssl/admin-kong-default.crt",
        "admin_ssl_cert_key_default": "/usr/local/kong/ssl/admin-kong-default.key",
        "admin_ssl_enabled": false,
        "anonymous_reports": false,
        "cassandra_consistency": "ONE",
        "cassandra_contact_points": [
            "127.0.0.1"
        ],
        "cassandra_data_centers": [
            "dc1:2",
            "dc2:3"
        ],
        "cassandra_keyspace": "kong",
        "cassandra_lb_policy": "RequestRoundRobin",
        "cassandra_port": 9042,
        "cassandra_repl_factor": 1,
        "cassandra_repl_strategy": "SimpleStrategy",
        "cassandra_schema_consensus_timeout": 60000,
        "cassandra_ssl": false,
        "cassandra_ssl_verify": false,
        "cassandra_timeout": 60000,
        "cassandra_username": "kong",
        "client_body_buffer_size": "64k",
        "client_max_body_size": "0",
        "client_ssl": false,
        "client_ssl_cert_default": "/usr/local/kong/ssl/kong-default.crt",
        "client_ssl_cert_key_default": "/usr/local/kong/ssl/kong-default.key",
        "database": "off",
        "db_cache_ttl": 0,
        "db_cache_warmup_entities": [
            "services",
            "plugins"
        ],
        "db_resurrect_ttl": 30,
        "db_update_frequency": 5,
        "db_update_propagation": 0,
        "declarative_config": "/etc/kong/yaml/kong.yaml",
        "dns_error_ttl": 1,
        "dns_hostsfile": "/etc/hosts",
        "dns_no_sync": false,
        "dns_not_found_ttl": 30,
        "dns_order": [
            "LAST",
            "SRV",
            "A",
            "CNAME"
        ],
        "dns_resolver": {},
        "dns_stale_ttl": 4,
        "enabled_headers": {
            "Server": true,
            "Via": true,
            "X-Kong-Proxy-Latency": true,
            "X-Kong-Upstream-Latency": true,
            "X-Kong-Upstream-Status": false,
            "latency_tokens": true,
            "server_tokens": true
        },
        "error_default_type": "text/plain",
        "headers": [
            "server_tokens",
            "latency_tokens"
        ],
        "kong_env": "/usr/local/kong/.kong_env",
        "loaded_plugins": {
            "file-log": true,
            "ip-restriction": true,
            "prometheus": true,
            "rate-limiting": true
        },
        "log_level": "notice",
        "lua_package_cpath": "",
        "lua_package_path": "./?.lua;./?/init.lua;",
        "lua_socket_pool_size": 30,
        "lua_ssl_verify_depth": 1,
        "mem_cache_size": "256m",
        "nginx_acc_logs": "/usr/local/kong/logs/access.log",
        "nginx_admin_directives": {},
        "nginx_conf": "/usr/local/kong/nginx.conf",
        "nginx_daemon": "on",
        "nginx_err_logs": "/usr/local/kong/logs/error.log",
        "nginx_http_directives": [
            {
                "name": "tcp_nodelay",
                "value": "on"
            },
            {
                "name": "log_format",
                "value": "json '{\"@timestamp\":\"$time_iso8601\",\"host\":\"$server_addr\",\"clientip\":\"$remote_addr\",\"http_x_forwarded_for\":\"$http_x_forwarded_for\",\"status\":\"$status\",\"request_method\":\"$request_method\",\"request_url\":\"$request_uri\",\"request_time\":\"$request_time\",\"size\":\"$body_bytes_sent\",\"upstream_time\":\"$upstream_response_time\",\"upstream_host\":\"$upstream_addr\",\"server_name\":\"$host\",\"uri\":\"$uri\",\"http_referer\":\"$http_referer\",\"http_user_agent\":\"$http_user_agent\"}'"
            },
            {
                "name": "keepalive_requests",
                "value": "\"1000\""
            },
            {
                "name": "keepalive_timeout",
                "value": "60s"
            },
            {
                "name": "lua_shared_dict",
                "value": "prometheus_metrics 100m"
            }
        ],
        "nginx_http_keepalive_requests": "1000",
        "nginx_http_keepalive_timeout": "60s",
        "nginx_http_log_format": "json '{\"@timestamp\":\"$time_iso8601\",\"host\":\"$server_addr\",\"clientip\":\"$remote_addr\",\"http_x_forwarded_for\":\"$http_x_forwarded_for\",\"status\":\"$status\",\"request_method\":\"$request_method\",\"request_url\":\"$request_uri\",\"request_time\":\"$request_time\",\"size\":\"$body_bytes_sent\",\"upstream_time\":\"$upstream_response_time\",\"upstream_host\":\"$upstream_addr\",\"server_name\":\"$host\",\"uri\":\"$uri\",\"http_referer\":\"$http_referer\",\"http_user_agent\":\"$http_user_agent\"}'",
        "nginx_http_lua_shared_dict": "prometheus_metrics 100m",
        "nginx_http_tcp_nodelay": "on",
        "nginx_kong_conf": "/usr/local/kong/nginx-kong.conf",
        "nginx_kong_stream_conf": "/usr/local/kong/nginx-kong-stream.conf",
        "nginx_optimizations": true,
        "nginx_pid": "/usr/local/kong/pids/nginx.pid",
        "nginx_proxy_directives": {},
        "nginx_sproxy_directives": {},
        "nginx_stream_directives": {},
        "nginx_worker_processes": "auto",
        "origins": {},
        "pg_database": "kong",
        "pg_host": "127.0.0.1",
        "pg_max_concurrent_queries": 0,
        "pg_port": 5432,
        "pg_semaphore_timeout": 60000,
        "pg_ssl": false,
        "pg_ssl_verify": false,
        "pg_timeout": 60000,
        "pg_user": "kong",
        "plugins": [
            "ip-restriction",
            "prometheus",
            "rate-limiting",
            "file-log"
        ],
        "prefix": "/usr/local/kong",
        "proxy_access_log": "/data/logs/kong/access.log json",
        "proxy_error_log": "/data/logs/kong/error.log",
        "proxy_listen": [
            "0.0.0.0:80",
            "0.0.0.0:443 ssl"
        ],
        "proxy_listeners": [
            {
                "http2": false,
                "ip": "0.0.0.0",
                "listener": "0.0.0.0:80",
                "port": 80,
                "proxy_protocol": false,
                "ssl": false,
                "transparent": false
            },
            {
                "http2": false,
                "ip": "0.0.0.0",
                "listener": "0.0.0.0:443 ssl",
                "port": 443,
                "proxy_protocol": false,
                "ssl": true,
                "transparent": false
            }
        ],
        "proxy_ssl_enabled": true,
        "real_ip_header": "X-Real-IP",
        "real_ip_recursive": "off",
        "router_consistency": "strict",
        "ssl_cert": "/usr/local/kong/ssl/kong-default.crt",
        "ssl_cert_csr_default": "/usr/local/kong/ssl/kong-default.csr",
        "ssl_cert_default": "/usr/local/kong/ssl/kong-default.crt",
        "ssl_cert_key": "/usr/local/kong/ssl/kong-default.key",
        "ssl_cert_key_default": "/usr/local/kong/ssl/kong-default.key",
        "ssl_cipher_suite": "custom",
        "ssl_ciphers": "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE:ECDH:AES:HIGH:!NULL:!aNULL:!MD5:!ADH:!RC4:!DH:!DHE",
        "ssl_preread_enabled": true,
        "stream_listen": [
            "off"
        ],
        "stream_listeners": {},
        "trusted_ips": {},
        "upstream_keepalive": 2048
    },
    "hostname": "bj5-techcenter-architecture-testkong-14",
    "lua_version": "LuaJIT 2.1.0-beta3",
    "node_id": "2717c619-3e64-434a-8950-3521a3acc1b9",
    "plugins": {
        "available_on_server": {
            "file-log": true,
            "ip-restriction": true,
            "prometheus": true,
            "rate-limiting": true
        },
        "enabled_in_cluster": [
            "prometheus"
        ]
    },
    "prng_seeds": {
        "pid: 5280": 182351922214,
        "pid: 5288": 200712968219,
        "pid: 5289": 481816710561,
        "pid: 5290": 116111481381,
        "pid: 5291": 215230103102,
        "pid: 5292": 892324921010,
        "pid: 5293": 162248200193,
        "pid: 5294": 344017514253,
        "pid: 5295": 165824021717,
        "pid: 5296": 842325563121,
        "pid: 5298": 631181891101,
        "pid: 5299": 211292171611,
        "pid: 5300": 201732361579,
        "pid: 5301": 191451134316,
        "pid: 5302": 231209920665,
        "pid: 5303": 223681521531,
        "pid: 5304": 652081796321,
        "pid: 5305": 170165391573,
        "pid: 5306": 217121232824,
        "pid: 5307": 481937915951,
        "pid: 5308": 144302325416,
        "pid: 5309": 231180443771,
        "pid: 5310": 188208162215,
        "pid: 5311": 184251252227,
        "pid: 5313": 376418735166,
        "pid: 5314": 181580861689,
        "pid: 5315": 272011804023,
        "pid: 5316": 196681241211,
        "pid: 5317": 822014320428,
        "pid: 5318": 362251812372,
        "pid: 5319": 281251921212,
        "pid: 5320": 386810119204,
        "pid: 5321": 180171115196
    },
    "tagline": "Welcome to kong",
    "timers": {
        "pending": 5,
        "running": 0
    },
    "version": "1.2.1"
}
- Operating system
CentOS Linux release 7.6.1810 (Core) 3.10.0-957.el7.x86_64 #1 SMP Thu Nov 8 23:39:32 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

**Comment by luxurine** [comment]
@thibaultcha  @hishamhm @bungle Hi, guys. We are going to use DB-less kong in the production environment, help please.

https://discuss.konghq.com/t/db-less-kong-1-2-1-config-reload-doesnt-take-effect/4014

**Comment by hishamhm** [comment]
@luxurine Thanks for the report! We are going to try to reproduce this on our end!

**Comment by luxurine** [comment]
> @luxurine Thanks for the report! We are going to try to reproduce this on our end!
Another problem is with admin api confif:

I have tried to benchmark with 300 targets(mock service), and when I use `http :8001/config config=kong.yml` to update, It comes out huge amount of error, most of them are "drop event" and "lock acquire timeout". After look up the source code,  I find there is no need to post crud create event every time for every entity in db-less mode (kong/db/declarative/init.lua) , as it will result in lock competition in on_target_event -> fetch_target_history -> mlcache.get -> run_callback lock. (as in db-less mode, there is no db, I'm not sure if it is necessary to use lock here )

```lua
  for _, entity_name in ipairs({"upstreams", "targets"}) do
    if entities[entity_name] then
      for _, item in pairs(entities[entity_name]) do
        post_crud_create_event(entity_name, item)
      end
    end
  end
```
--->
```lua
for _, entity_name in ipairs({"upstreams", "targets"}) do
    if entities[entity_name] then
      -- one event per upstream
      local upstreamid_target = {}
      for _, item in pairs(entities[entity_name]) do
        if entity_name == "targets" then
	  if not upstreamid_target[item.upstream.id] then
            post_crud_create_event(entity_name, item)
	  end
	  upstreamid_target[item.upstream.id] = item
	else
          post_crud_create_event(entity_name, item)
 	end 
      end
    end
  end
```

More over, balacer will be clear and rebuild, in production, it will cause http 500 error as it seems not that graceful.

**Comment by guanlan** [comment]
@luxurine It should be fixed with the 1.3 release, closing this.
